src/helics/core/FederateState.cpp

```diff
@@ -833,6 +835,30 @@ void FederateState::finalize()
 #endif
 }
 
+void FederateState::processCommunications(std::chrono::milliseconds period)
+{
+    ActionMessage treq(CMD_USER_RETURN);
+    treq.source_id = global_id.load();
+    // the user return should only be for this thread, other threads
+    treq.messageID = static_cast<int32_t>(std::hash<std::thread::id>{}(std::this_thread::get_id()));
+    addAction(treq);
+    auto starttime = std::chrono::steady_clock::now();
+    auto ret = MessageProcessingResult::CONTINUE_PROCESSING;
+    while (ret != MessageProcessingResult::USER_RETURN) {
+        ret = genericUnspecifiedQueueProcess(true);
+        if (ret == MessageProcessingResult::BUSY) {
+            return;
+        }
+    }
+    if (period > std::chrono::milliseconds(0)) {
+        auto ctime = std::chrono::steady_clock::now();
+        if (period - (ctime - starttime) > std::chrono::milliseconds(10)) {
+            std::this_thread::sleep_for(period - (ctime - starttime));
+            processCommunications(std::chrono::milliseconds(0));
```

Comment:
Is this just trying to consume time so that the processCommunications function returns within 10ms of the user requested time period? Is the period supplied a lower bound on the amount of time that will be spent processing communications?